src/Aquifer.Well.API/Services/NotificationHubService.cs

```
@@ -0,0 +1,170 @@
+using Microsoft.Extensions.Options;
+using Microsoft.Azure.NotificationHubs;
+
+namespace Aquifer.Well.API.Services;
+
+public class NotificationHubService : INotificationService
+{
+    private readonly NotificationHubClient _hub;
+    private readonly Dictionary<string, NotificationPlatform> _installationPlatform;
+    private readonly ILogger<NotificationHubService> _logger;
+    public NotificationHubService(IOptions<NotificationHubOptions> options, ILogger<NotificationHubService> logger)
+
+    {
+        _logger = logger;
+        _hub = NotificationHubClient.CreateClientFromConnectionString(options.Value.ConnectionString, options.Value.Name);
+
+        _installationPlatform = new Dictionary<string, NotificationPlatform>
+        {
+            { nameof(NotificationPlatform.Apns).ToLower(), NotificationPlatform.Apns },
+            { nameof(NotificationPlatform.FcmV1).ToLower(), NotificationPlatform.FcmV1 }
+        };
+    }
+
+    public async Task<bool> CreateOrUpdateInstallationAsync(DeviceInstallation deviceInstallation, CancellationToken token)
+    {
+        if (string.IsNullOrWhiteSpace(deviceInstallation?.InstallationId) ||
+            string.IsNullOrWhiteSpace(deviceInstallation?.Platform) ||
+            string.IsNullOrWhiteSpace(deviceInstallation?.PushChannel))
+        {
+            return false;
+        }
+
+        var installation = new Installation()
+        {
+            InstallationId = deviceInstallation.InstallationId,
+            PushChannel = deviceInstallation.PushChannel,
+            Tags = deviceInstallation.Tags
+        };
+
+        if (_installationPlatform.TryGetValue(deviceInstallation.Platform, out var platform))
+        {
+            installation.Platform = platform;
+        }
+        else
+        {
+            return false;
+        }
+
+        try
+        {
+            await _hub.CreateOrUpdateInstallationAsync(installation, token);
+        }
+        catch
+        {
+            return false;
```

Do we need to know what happened here? This swallows the exception, so we'd have no logging or anything on it.